packages/esm-patient-test-results-app/src/print/print.component.tsx

```diff
@@ -0,0 +1,257 @@
+  const subheader = t('testReasults', 'Test Results');
+
+  const tableHeaders = [
+    { key: 'testType', header: 'Test Type' },
+    { key: 'date', header: 'Date' },
+    { key: 'result', header: 'Result' },
+    { key: 'normalRange', header: 'Normal Range' },
+  ];
+
+  const fromDate = (date) => {
+    setSelectedFromDate(date);
+  };
+
+  const toDate = (date) => {
+    setSelectedToDate(date);
+  };
+
+  const handlePrint = useReactToPrint({
+    content: () => componentRef.current,
+  });
+
+  const patient = usePatient(patientUuid);
+
+  const patientDetails = useMemo(() => {
+    const getGender = (gender: string): string => {
+      switch (gender) {
+        case 'male':
+          return t('male', 'Male');
+        case 'female':
+          return t('female', 'Female');
+        case 'other':
+          return t('other', 'Other');
+        case 'unknown':
+          return t('unknown', 'Unknown');
+        default:
+          return gender;
+      }
+    };
+
+    const identifiers =
+      patient?.patient?.identifier?.filter(
+        (identifier) => !excludePatientIdentifierCodeTypes?.uuids.includes(identifier.type.coding[0].code),
+      ) ?? [];
+
+    return {
+      name: `${patient?.patient?.name?.[0]?.given?.join(' ')} ${patient?.patient?.name?.[0].family}`,
+      age: age(patient?.patient?.birthDate),
+      gender: getGender(patient?.patient?.gender),
+      location: patient?.patient?.address?.[0].city,
+      identifiers: identifiers?.length ? identifiers.map(({ value, type }) => value) : [],
+    };
+  }, [patient, t, excludePatientIdentifierCodeTypes?.uuids]);
+
+  useEffect(() => {}, [selectedFromDate, selectedToDate]);
+  const testResults = useMemo(() => {
+    if (selectedFromDate && selectedToDate) {
+      const selectedDateObj = new Date(selectedFromDate);
+      const selectedToDateObj = new Date(selectedToDate);
+
+      return panels
+        .filter((panel) => {
+          const panelDate = new Date(panel.effectiveDateTime);
+          return panelDate >= selectedDateObj && panelDate <= selectedToDateObj;
+        })
+        .map((panel) => {
+          const panelDate = new Date(panel.effectiveDateTime);
+          const formattedPanelDate = panelDate.toLocaleDateString();
+          return {
+            id: panel.id,
+            testType: panel.name,
+            date: formattedPanelDate,
+            result: panel.value,
+            normalRange: panel?.meta?.range ?? '--',
+          };
+        });
+    }
+
+    return panels.map((panel) => {
+      const panelDate = new Date(panel.effectiveDateTime);
+      const formattedPanelDate = panelDate.toLocaleDateString();
+      return {
+        id: panel.id,
+        testType: panel.name,
+        date: formattedPanelDate,
+        result: panel.value,
+        normalRange: panel?.meta?.range ?? '--',
+      };
+    });
+  }, [panels, selectedFromDate, selectedToDate]);
+
+  return (
+    <div>
```

Comment:
Dunno if you've considered it, but it might be nice to show a `ModalHeader` with the text `Print test results` or similar just above the `ModalBody` component.

Reply:
looks like this was added